### PR TITLE
GoogleMapsCard common component

### PR DESCRIPTION
**Pull request by ANITA-0604**
Created a GoogleMapsCard component to show google maps.

Props: 
```
  data?: DirectionsServiceProps["options"];
  width?: string; // default: "100%"
  height?: string; // default: "300px"
  draggable?: boolean; // default: false
```

Example `data`: 
```
import type { DirectionsServiceProps } from "@react-google-maps/api";
{
  origin: " 新北市政府",
  waypoints: [{ location: "中正紀念堂" }, { location: "國父紀念館" }], // 中途停靠站 (Rider)
  destination: "台北市政府",
  travelMode: "DRIVING" as DirectionsServiceProps["options"]["travelMode"] ,
}
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,6 +23,10 @@
     "next": "13.2.4",
     "react": "18.2.0",
     "react-dom": "18.2.0",
+    "react-icons": "^4.8.0",
     "typescript": "5.0.3"
+  },
+  "devDependencies": {
+    "@types/google.maps": "^3.53.1"
```
Comment (newb1er) [inline package.json:30]: Is the type definition lib correct?
I think it should be [`@types/google-map-react`](https://www.npmjs.com/package/@types/google-map-react)

File: tsconfig.json [hunk 2]
```diff
@@ -16,7 +16,8 @@
     "incremental": true,
     "paths": {
       "@/*": ["./src/*"]
-    }
+    },
+    "types": ["google.maps"]
```
Comment (newb1er) [inline tsconfig.json:20]: If the type definition lib is correct, then there's no need to set this config.

File: src/modules/components/GoogleMapsCard.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  GoogleMap,
+  useLoadScript,
+  DirectionsService,
+  DirectionsRenderer,
+  DirectionsServiceProps,
+} from "@react-google-maps/api";
+import { useRef, useState } from "react";
+
+interface Props {
+  data?: any;
```
Comment (newb1er) [inline src/modules/components/GoogleMapsCard.tsx:12]: Maybe the type of `data` will be found in [`@types/google-map-react`](https://www.npmjs.com/package/@types/google-map-react)

**Review by newb1er — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (ANITA-0604) [reply package.json:30]: Yes, it's correct. 
`@types/google-map-react` is the type for another repo (google-map-react)

Comment (newb1er) [reply package.json:30]: Oh i see

**Review by newb1er — APPROVED** [review]

File: src/modules/components/GoogleMapsCard.tsx (hunk 3 above)
Comment (ANITA-0604) [reply src/modules/components/GoogleMapsCard.tsx]: [chore: correct Google Maps DirectionService type](https://github.com/CoRide-tw/frontend/pull/2/commits/d21a95436c9b8b990a9de47f083b684809cbe72a) fixed!!

**Review by ericyangchen — APPROVED** [review]

**Comment by ANITA-0604** [comment]
When you receiving data from API, plz remember to cast `travelMode` to `DirectionsServiceProps["options"]["travelMode"]`.
Example can be found in the description.

**Review by ericyangchen — APPROVED** [review]

**Pull request merged**